### PR TITLE
stop loading messages when packages are attached

### PR DESCRIPTION
**Pull request by topepo**
Since `foreach` attaches packages.

### CODE REVIEW

**Comment by topepo** [comment]
Should we do this?

**Comment by juliasilge** [comment]
So I am in favor of this change, because I think that all that ⬇️ about attaching rlang, vctrs, rpart is not helpful or normal to the user. My opinion is that users don't expect that level of detail in messages about what underlying packages are being used.

``` r
library(tidymodels)
#> ── Attaching packages ────────────────────────────────────── tidymodels 0.1.2 ──
#> ✓ broom     0.7.3      ✓ recipes   0.1.15
#> ✓ dials     0.0.9      ✓ rsample   0.0.8 
#> ✓ dplyr     1.0.2      ✓ tibble    3.0.4 
#> ✓ ggplot2   3.3.3      ✓ tidyr     1.1.2 
#> ✓ infer     0.5.3      ✓ tune      0.1.2 
#> ✓ modeldata 0.1.0      ✓ workflows 0.2.1 
#> ✓ parsnip   0.1.4      ✓ yardstick 0.0.7 
#> ✓ purrr     0.3.4
#> ── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
#> x purrr::discard() masks scales::discard()
#> x dplyr::filter()  masks stats::filter()
#> x dplyr::lag()     masks stats::lag()
#> x recipes::step()  masks stats::step()
library(themis)
#> Registered S3 methods overwritten by 'themis':
#>   method                  from   
#>   bake.step_downsample    recipes
#>   bake.step_upsample      recipes
#>   prep.step_downsample    recipes
#>   prep.step_upsample      recipes
#>   tidy.step_downsample    recipes
#>   tidy.step_upsample      recipes
#>   tunable.step_downsample recipes
#>   tunable.step_upsample   recipes
#> 
#> Attaching package: 'themis'
#> The following objects are masked from 'package:recipes':
#> 
#>     step_downsample, step_upsample
library(baguette)

data(okc)

okc_folds <- vfold_cv(okc, v = 3)

rec <- recipe(Class ~ age + diet + height, data = okc) %>%
  step_downsample(diet)

bag_spec <- bag_tree() %>%
  set_engine("rpart", times = 15) %>%
  set_mode("classification")

bag_wf <- workflow() %>%
  add_recipe(rec) %>%
  add_model(bag_spec)

doParallel::registerDoParallel()
fit_resamples(bag_wf, okc_folds)
#> 
#> Attaching package: 'rlang'
#> The following objects are masked from 'package:purrr':
#> 
#>     %@%, as_function, flatten, flatten_chr, flatten_dbl, flatten_int,
#>     flatten_lgl, flatten_raw, invoke, list_along, modify, prepend,
#>     splice
#> 
#> Attaching package: 'vctrs'
#> The following object is masked from 'package:tibble':
#> 
#>     data_frame
#> The following object is masked from 'package:dplyr':
#> 
#>     data_frame
#> 
#> Attaching package: 'rpart'
#> The following object is masked from 'package:dials':
#> 
#>     prune
#> # Resampling results
#> # 3-fold cross-validation 
#> # A tibble: 3 x 4
#>   splits              id    .metrics         .notes          
#>   <list>              <chr> <list>           <list>          
#> 1 <split [39.9K/20K]> Fold1 <tibble [2 × 4]> <tibble [0 × 1]>
#> 2 <split [39.9K/20K]> Fold2 <tibble [2 × 4]> <tibble [0 × 1]>
#> 3 <split [39.9K/20K]> Fold3 <tibble [2 × 4]> <tibble [0 × 1]>
```

<sup>Created on 2021-01-04 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Review by juliasilge — APPROVED** [review]

File: R/grid_code_paths.R [hunk 1]
```diff
@@ -24,45 +24,49 @@ tune_grid_loop <- function(resamples, grid, workflow, metrics, control, rng) {
...
-    }
-  } else if (identical(parallel_over, "everything")) {
-    results <- foreach::foreach(
-      iteration = iterations,
-      .packages = packages,
-      .errorhandling = "pass"
-    ) %:%
-      foreach::foreach(
-        row = rows,
+    suppressPackageStartupMessages(
```
Comment (DavisVaughan) [inline R/grid_code_paths.R:27]: I don't see why you need `suppressPackageStartupMessages()` here. Shouldn't it _only_ be used with `load_pkgs()` and `load_namespace()` in `tune_grid_loop_iter()`?

Comment (topepo) [reply R/grid_code_paths.R:27]: We still need it since we use `foreach(.packages = packages)`. That doesn't use the loading functions in `tune`

Comment (DavisVaughan) [reply R/grid_code_paths.R:27]: 😭

**Pull request merged**